src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -475,9 +476,34 @@ CreateHashDistributedTableShards(Oid relationId, Oid colocatedTableId,
 	 */
 	EnsureSchemaExistsOnAllNodes(relationId);
 
+	/*
+	 * Decide whether to use exclusive connections per placement or not. Note that
+	 * if the local table is not empty, we cannot use sequential mode since the COPY
+	 * operation that'd load the data into shards currently requires exclusive
+	 * connections.
+	 */
 	if (RegularTable(relationId))
 	{
-		useExclusiveConnection = IsTransactionBlock() || !localTableEmpty;
+		if (!localTableEmpty && MultiShardConnectionType == SEQUENTIAL_CONNECTION)
```

Comment:
As we've discussed in person, @furkansahin is going to move this logic to a separate function in the next PR, seems acceptable for now.